pkg/apis/build/v1alpha1/image_builds.go

```diff
@@ -57,11 +57,23 @@ func (im *Image) buildNeeded(lastBuild *Build, sourceResolver *SourceResolver, b
 		reasons = append(reasons, BuildReasonBuildpack)
 	}
 
-	if lastBuild.Status.RunImage != "" && lastBuild.Status.RunImage != builder.RunImage() {
-		reasons = append(reasons, BuildReasonStack)
+	if lastBuild.Status.RunImage != "" {
```

Comment:
It would be great to move much of this logic to the build itself in a future refactor. 

Something like: 

```
lastBuild.MatchesStack(builder)
```